api/files.py

```diff
@@ -34,9 +34,20 @@ def getHashingFieldStorage(upload_dir, hash_alg):
     class HashingFieldStorage(cgi.FieldStorage):
         bufsize = 2**20
         def make_file(self, binary=None):
+            if not self.filename:
+                return self.file
             self.open_file = HashingFile(os.path.join(upload_dir, os.path.basename(self.filename)), hash_alg)
             return self.open_file
 
+        # override private method __write of superclass FieldStorage
+        # _FieldStorage__file is the private variable __file of the same class
+        def _FieldStorage__write(self, line):
+            if self._FieldStorage__file is not None:
```

Comment:
1. `self.__write` is private method and would be translated internally by Python to `self._HashingFieldStorage__write` and wouldn't have any effect
2. `self.__file` has the same behavior
3. This was my question! However check the original `__write` method and you will see that the override is safe, but maybe there is a better place for this logic to go.